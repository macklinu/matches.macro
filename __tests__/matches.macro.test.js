import pluginTester from 'babel-plugin-tester'
import plugin from 'babel-plugin-macros'
import { stripIndent, stripIndents } from 'common-tags'

pluginTester({
  plugin,
  pluginName: 'matches.macro',
  snapshot: true,
  babelOptions: {
    filename: __filename,
  },
  tests: [
    {
      title: 'matches() throws without any arguments',
      error: true,
      code: stripIndents`
    import matches from '../src/matches.macro'
    
    const isBasketball = matches()
    `,
    },
    {
      title: 'matches() throws with non-object argument',
      error: true,
      code: stripIndents`
    import matches from '../src/matches.macro'
    
    const isBasketball = matches('hello')
    `,
    },
    stripIndent`
    import matches from '../src/matches.macro'
    
    const isNbaBasketball = matches({
      type: 'basketball',
      'ballColor': 'orange',
      ballMaker: /spalding/i,
      'ballMaker': /spalding/i,
      'league.type': 'NBA'
    })
    `,
  ],
})
