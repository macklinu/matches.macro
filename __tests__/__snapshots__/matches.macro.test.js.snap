// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches.macro 1. matches.macro: 1. matches.macro 1`] = `

import matches from '../src/matches.macro'

const isNbaBasketball = matches({
  type: 'basketball',
  'ballColor': 'orange',
  'league.type': 'NBA'
})

      ↓ ↓ ↓ ↓ ↓ ↓

const isNbaBasketball = arg => {
  if (arg?.type === 'basketball') return true
  if (arg?.ballColor === 'orange') return true
  if (arg?.league?.type === 'NBA') return true
  return false
}


`;

exports[`matches.macro matches() throws with non-object argument: matches() throws with non-object argument 1`] = `

import matches from '../src/matches.macro'

const isBasketball = matches('hello')

      ↓ ↓ ↓ ↓ ↓ ↓

MacroError: /Users/macklin.underdown/dev/matches.macro/__tests__/matches.macro.test.js: matches() must be called as a function with one object argument

`;

exports[`matches.macro matches() throws without any arguments: matches() throws without any arguments 1`] = `

import matches from '../src/matches.macro'

const isBasketball = matches()

      ↓ ↓ ↓ ↓ ↓ ↓

MacroError: /Users/macklin.underdown/dev/matches.macro/__tests__/matches.macro.test.js: matches() must be called as a function with one object argument

`;
